### PR TITLE
Implemented error handling for Bedrock

### LINKED ISSUE #2295: AWS Bedrock Verify and update error handling implementation
### Describe the story, please be clear on scope of the story.
A stub exists in the InvokeModelAsyncWrapper as HandleError. This passes the exception to RecordLlmEvent for passing to NoticeError- if this doesn't work well another approach will need to be determined:

- http.statusCode: Status code returned from error response
- Error.code: Value of code attribute returned from exception body
- Error.param: Value of param attribute returned from exception body
- Completion_id: Agent-generated unique ID for chat completion creation - does not exist for embedding errors
- Embedding_id: Agent-generated unique ID for embedding creation - does not exist for chat completion errors

The test app includes a model that creates an error for manual testing.

### Acceptance Criteria

- Error is captured with correct attributes.
- Confirm agent works with test app

### PR DESCRIPTION
**Pull request by tippmar-nr**
Error handling for Bedrock, based on requirements in the spec. Generates error events similar to 

```
["BW7NK7CFbFXrAAQqrgglOVll6NSgAAQBIQArJQEAAFys5SnXZAIECBONVAMACjEwLjIxLjAuMzQACU5SREQ1WlZUMwAVQUkgTW9uaXRvcmluZyBTaWduYWxS", {
        "reservoir_size": 8,
        "events_seen": 1
    }, [[{
                "duration": 19.778362,
                "error.class": "Custom Error",
                "error.message": "You don't have access to the model with the specified model ID.",
                "externalCallCount": 1.0,
                "externalDuration": 0.6889039874076843,
                "guid": "3b38fb9cbda92a6d",
                "priority": 1.2292109727859497,
                "sampled": true,
                "spanId": "81f43eaa26391287",
                "timestamp": 1709757514125,
                "traceId": "930d051d14fe639dba54c70ec91025a8",
                "transactionName": "WebTransaction/ASP/chathub",
                "type": "TransactionError"
            }, {
                "completion_id": "2c606da9-4262-4dca-bc75-ed8ffc0937e0",
                "error.code": "AccessDeniedException",
                "http.statusCode": "403",
                "llm.conversation_id": "a9028608-c5f6-4c78-ab16-7e653361c721",
                "llm.SomethingAwesome": 42,
                "NotAwesome": 13
            }, {
                "host.displayName": "NRDD5ZVT3",
                "http.statusCode": 101,
                "request.headers.accept-encoding": "gzip, deflate, br, zstd",
                "request.headers.accept-language": "en-US,en;q=0.9",
                "request.headers.cache-control": "no-cache",
                "request.headers.connection": "Upgrade",
                "request.headers.host": "localhost:5001",
                "request.headers.origin": "https://localhost:5001",
                "request.headers.pragma": "no-cache",
                "request.headers.sec-websocket-extensions": "permessage-deflate; client_max_window_bits",
                "request.headers.sec-websocket-key": "c70e2PolE06sEaSaDwx7yg==",
                "request.headers.sec-websocket-version": "13",
                "request.headers.upgrade": "websocket",
                "request.headers.user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36",
                "request.method": "GET",
                "request.uri": "/chathub",
                "response.status": "101"
            }
        ]]]
```

Closes  #2295

### CODE REVIEW

**Review by chynesNR — APPROVED** [review]

File: src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Bedrock/InvokeModelAsyncWrapper.cs [hunk 1]
```diff
@@ -60,25 +61,31 @@ void InvokeTryProcessResponse(Task responseTask)
                 if (responseTask.IsFaulted)
                 {
                     HandleError(segment, invokeModelRequest, responseTask, agent);
+                    return;
```
Comment (nrcventura) [inline src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Bedrock/InvokeModelAsyncWrapper.cs:64]: This seems to return before ending the segment.

Comment (tippmar-nr) [reply src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Bedrock/InvokeModelAsyncWrapper.cs:64]: I originally ended the segment at the top of this method, but that seemed to result in no errors getting noticed. So I guess I should end it here just before returning?

File: src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Llm/EventHelper.cs [hunk 2]
```diff
@@ -57,6 +58,20 @@ public static class EventHelper
 
             agent.RecordLlmEvent("LlmChatCompletionSummary", attributes);
 
+            if (isError)
```
Comment (jaffinito) [inline src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Llm/EventHelper.cs:61]: Interesting approach to getting the error attributes.  I was attempting to get them later on, but I think this works!

File: src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Bedrock/InvokeModelAsyncWrapper.cs (hunk 1 above)
Comment (nrcventura) [reply src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Bedrock/InvokeModelAsyncWrapper.cs:64]: Yes, I think we should end it here before returning. I don't think there's a cleaner way.

**Review by jaffinito — APPROVED** [review]

**Review by chynesNR — APPROVED** [review]
Looks good!

**Pull request merged**